src/feed-store-provider.c

```diff
@@ -85,37 +90,44 @@ call_dbus_proxy_and_construct_from_models_and_shards (GDBusProxy
                      model_variant);
 
   /* Now that we have the models and shards, we can marshal them into
-   * a GPtrArray containing the discovery-feed models */
-  return marshal_func ((const gchar * const *) shards_strv,
-                       model_props_variants,
-                       marshal_data);
+   * a GSList containing the discovery-feed models */
```

Comment:
It looks like this is still a `GPtrArray` to me...

Reply:
The `model_props_variants` is a `GPtrArray` but the props get marshalled into objects in a `GSList` by `ModelsFromResultsAndShardsFunc`